APE14.rst

```diff
@@ -242,6 +245,14 @@ low-level API recommends:
             ``y`` is the vertical coordinate.
             """
 
+        def index_to_world_values(self, *index_arrays):
+            """
+            Convert array indices to world coordinates. This is the same as
+            ``pixel_to_world_values`` except that the indices should be given
```

Comment:
Here (and the same below in ``world_to_index_values``: I wonder if this needs to be clear about something like "the native order of the data array" instead of specifying the row-major way of describing it?  That is, if at some point in the future, a new array package comes out that is column-major instead of row-major, presumably we want this to work that such that indexing is correct, right?

This could just be a single small addition - e.g., ``in `(i, j)` order`` -> ``in the appropriate index order for the data - for numpy, `(i, j)` order``.  But then in the added discussion below, there should be mention that the real "contract" for the ``index`` methods is that they *match* the data, regardless as to whether that's row- or column-major

Reply:
In principle I agree but there is an additional complication here - it's possible to define a WCS not attached to a specific array - and in fact the data is not a property of the WCS. So a single WCS could apply to both data in Numpy form and in another form. So for that reason I actually feel like we'd be better off just renaming the methods to say ``numpy_index`` instead of just ``index`` and then one day if another popular package comes along, we can add new methods for that.

Reply:
See the change I suggest in ef717cb0972f312c90a9ecb927e9f6d4ffbf4566